sklearn/tree/_utils.pyx

```diff
@@ -120,7 +121,7 @@ cdef class Stack:
 
     cdef int push(self, SIZE_t start, SIZE_t end, SIZE_t depth, SIZE_t parent,
                   bint is_left, double impurity,
-                  SIZE_t n_constant_features) nogil:
+                  SIZE_t n_constant_features) nogil except *:
```

Comment:
In the current API -1 is only returned on memory errors, let's use that to reduce the overhead: `except -1`.